packages/dds/tree/src/core/rebase/utils.ts

```diff
@@ -232,6 +243,8 @@ export function rebaseBranch<TChange>(
 				targetCommits,
 				sourceCommits,
 			},
+			sourceBranchLength: sourcePath.length, // The length of the sourcePath should be EQUAL to the length of the targetPath.
+			rebaseDistance: 0,
```

Comment:
I think the best approach would be to return 3 numbers instead of 2:
- sourceBranchLenght (same as before)
- rebaseDistance (this should always be `targetCommitIndex+1`)
- countInCommon (sourcePath.length - sourceSet.size)